src/Features/CSharp/Portable/ReplaceMethodWithProperty/CSharpReplaceMethodWithPropertyService.cs

```diff
@@ -0,0 +1,272 @@
+﻿using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.ReplaceMethodWithProperty;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.ReplaceMethodWithProperty
+{
+    [ExportLanguageService(typeof(IReplaceMethodWithPropertyService), LanguageNames.CSharp), Shared]
+    internal class CSharpReplaceMethodWithPropertyService : IReplaceMethodWithPropertyService
+    {
+        public string GetMethodName(SyntaxNode methodNode)
+        {
+            return ((MethodDeclarationSyntax)methodNode).Identifier.ValueText;
+        }
+
+        public SyntaxNode GetMethodDeclaration(SyntaxToken token)
+        {
+            var containingMethod = token.Parent.FirstAncestorOrSelf<MethodDeclarationSyntax>();
+            if (containingMethod == null)
+            {
+                return null;
+            }
+
+            var start = containingMethod.AttributeLists.Count > 0
+                ? containingMethod.AttributeLists.Last().GetLastToken().GetNextToken().SpanStart
+                : containingMethod.SpanStart;
+
+            // Offer this refactoring anywhere in the signature of the method.
+            var position = token.SpanStart;
+            if (position < start || position > containingMethod.ParameterList.Span.End)
+            {
+                return null;
+            }
+
+            return containingMethod;
+        }
+
+        public SyntaxNode ConvertMethodsToProperty(
+            SemanticModel semanticModel,
+            SyntaxGenerator generator, GetAndSetMethods getAndSetMethods,
+            string propertyName, bool nameChanged)
+        {
+            var getMethodDeclaration = getAndSetMethods.GetMethodDeclaration as MethodDeclarationSyntax;
+            if (getMethodDeclaration == null)
+            {
+                return getMethodDeclaration;
+            }
+
+            var setMethod = getAndSetMethods.SetMethodDeclaration as MethodDeclarationSyntax;
+
+            var getAccessor = CreateGetAccessor(getAndSetMethods);
+            var setAccessor = CreateSetAccessor(semanticModel, generator, getAndSetMethods);
+
+            var accessorList = SyntaxFactory.AccessorList(SyntaxFactory.SingletonList(getAccessor));
+            if (setAccessor != null)
+            {
+                accessorList = accessorList.AddAccessors(new[] { setAccessor });
+            }
+
+            var property = SyntaxFactory.PropertyDeclaration(
```

Comment:
No.  I just copy all the trivia over.  I leave it to the user to decide what they want to do with it.